hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1835,36 +1846,49 @@ func (r *HostedClusterReconciler) reconcileIgnitionServer(ctx context.Context, c
 								"--key-file", "/var/run/secrets/ignition/serving-cert/tls.key",
 								"--registry-overrides", convertRegistryOverridesToCommandLineFlag(r.ReleaseProvider.GetRegistryOverrides()),
 								"--platform", string(hcluster.Spec.Platform.Type),
+								"--work-dir", "/payloads",
 							},
 							LivenessProbe: &corev1.Probe{
 								ProbeHandler: corev1.ProbeHandler{
-									TCPSocket: &corev1.TCPSocketAction{
-										Port: intstr.FromInt(9090),
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
```

Comment:
what was the rationale to change this?

Reply:
So this also reveals yet another mistake I made, which is that the new probe setup is incompatible with an N-1 control plane, which won't be serving health/ready on the newly introduced ports, which means the pods will never be ready (and thus accessible). Doing this sort of structural change would require some more thought. I'll just revert it for now, as what is there before basically worked, but does generate some log spam from the TLS errors. We can just consider that a separate issue, not required for this work. Thanks for the review!